16.0 mig commown factor out commown partner set invoice recipient

Honeyxilia opened the pull request.

Honeyxilia committed:
Create commown_partner_set_invoice_recipent

This module has a goal to set a partner as the recipent of their parent partner's contracts invoices.

Honeyxilia committed:
Moved code out of commown into commown_partner_set_invoice_recipent

Honeyxilia committed:
Adapted to account.move change

Honeyxilia committed:
Added non-breaking space next to text for better rendering

Honeyxilia committed:
[test] Removed commown_payment_token_uniquify test class dependency

Since we only depended on commown_payment_token_uniquify for its test class, especially it's setUpClass method,
we made this module self-sufficient by redeclaring the necessary data to perform the action test.

fcayre commented on commown_partner_set_invoice_recipient/tests/test_res_partner.py:
```diff
@@ -1,16 +1,69 @@
+
+          * first token is the main one of the contract's partner
+          * second token is directly linked to the contract
+        """
```
La docstring est obsolète (par exemple on ne crée plus qu'un token de paiement ici).

fcayre reviewed (dismissed):
Juste une dosctring à revoir et go pour la fusion !

Honeyxilia committed:
[test] Added inject_payment_data call if account_invoice_merge_auto_pay is installed

The account_invoice_merge_auto_pay module adds a mechanism to automatically merge and pay invoices,
which is crashes if the auto_merge attribute is set to True (set in the contract_auto_merge_invoice module)
and the payment mode of the invoice is not set.
However, depending on the used modules, there might be no payment mode associated with the environnement company.
As such, when this module is installed, we call the inject_payment_data method to automatically create a payment mode for the enrivonnement company, and the company partner we create.

Honeyxilia commented:
/ocatbot merge patch

Honeyxilia commented:
/ocabot merge patch
